packages/core/src/util/url.js

```diff
@@ -7,11 +7,22 @@
 
 const secrets = require('../secrets');
 
+const embeddedCredentialsRegex = /^(https?:\/\/)[^@:?/#;]+:[^@]+@(.*)$/i;
+
 /**
  * @param {string} url
  * @returns {string}
  */
-exports.discardUrlParameters = function discardUrlParameters(url) {
+exports.sanitizeUrl = function sanitizeUrl(url) {
+  // Remove embedded credentials like the user:password part in https://user:password@example.org.
+  // Do RegExp#test before doing RegExp#exec because a non-match is the hot path.
+  if (embeddedCredentialsRegex.test(url)) {
```

Comment:
Personally, in general I would go for the standard lib URL module when it comes to parse URL.
But I don't spot anything wrong with the RegExp either, so the solution looks great. It takes me a moment to understand the regexp, but that's all :)

Reply:
Oh, yes, that's absolutely right. Don't know why I didn't think of that before. I'll rebuild this with `url`, this will be much safer/less error-prone 👍